met-web/src/components/survey/building/index.tsx

```diff
@@ -109,6 +110,17 @@ const SurveyFormBuilder = () => {
         return <FormBuilderSkeleton />;
     }
 
+    const hasEngagement = !!savedSurvey.engagement;
+    const isEngagementDraft = savedSurvey.engagement?.status_id == EngagementStatus.Draft;
+
+    if (hasEngagement && !isEngagementDraft) {
```

Comment:
Cool. I think the react pattern is to add stuff like this in a  useEffect which kinda is like the hook version of ComponentDidMount life cycle methods. You want to run that if the component mounted and depending on the savedSurvey field so I think useEffect(() => {... that block}, [savedSurvey])

Reply:
Noted & Changed, thanks!